src/tests/roc_audio/test_resampler.cpp

```diff
@@ -100,147 +104,176 @@ TEST_GROUP(resampler) {
 TEST(resampler, invalid_scaling) {
     enum { ChMask = 0x1, InvalidScaling = FrameSize };
 
-    MockReader reader;
-    ResamplerReader rr(reader, buffer_pool, allocator, config, ChMask, FrameSize);
+    for (size_t n_backend = 0; n_backend < Test_n_resampler_backends; n_backend++) {
+        ResamplerBackend backend = Test_resampler_backends[n_backend];
 
-    CHECK(rr.valid());
+        MockReader reader;
+        core::UniquePtr<IResampler> resampler(
+            resampler_map.new_resampler(backend, allocator, config, ChMask, FrameSize),
+            allocator);
```

Comment:
Let's also add a check that resampler was successfully constructed (to all tests):

```
core::UniquePtr<IResampler> resampler(...);
CHECK(resampler);
```